src/CLI/clitree/cli-xml/tam.xml

```diff
@@ -1,133 +1,93 @@
-    <ACTION builtin="clish_nop"></ACTION>
-    </COMMAND>
-  </VIEW>
-  <VIEW name="configure-tam-view"
-        prompt="${SYSTEM_NAME}(config-tam)# "
-        depth="2">
+<!--
+Copyright 2019 Dell, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
 
-        <NAMESPACE ref="configure-view"
-                help="false"
-                completion="false"/>
+    http://www.apache.org/licenses/LICENSE-2.0
 
-         <COMMAND name="no"
-              help="Negate a command or set its defaults">
-         </COMMAND>
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<CLISH_MODULE
+    xmlns="http://www.dellemc.com/sonic/XMLSchema"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:xi="http://www.w3.org/2001/XInclude"
+    xsi:schemaLocation="http://www.dellemc.com/sonic/XMLSchema
+                        http://www.dellemc.com/sonic/XMLSchema/clish.xsd"
+    >
+<!--=========================================================================-->
+
+  <VIEW name="enable-view">
+    <COMMAND name="show tam" help="Show TAM info"/>
+    <COMMAND name="show tam device" help="Show TAM device info" mode="subcommand" ptype="SUBCOMMAND">
+      <ACTION>python $SONIC_CLI_ROOT/sonic-cli-tam.py get_sonic_tam_sonic_tam_tam_device_table show_tam_device.j2</ACTION>
+    </COMMAND>
+    <COMMAND name="show tam collector" help="Show TAM collector info" mode="subcommand" ptype="SUBCOMMAND">
+      <PARAM name="name" help="Name of the collector" ptype="STRING_32" optional="true"> </PARAM>
+      <ACTION>
+        if test "${name}" = "all" ; then&#xA;
+          python $SONIC_CLI_ROOT/sonic-cli-tam.py get_sonic_tam_sonic_tam_tam_collector_table show_tam_collector.j2&#xA;
+        elif test -z "${name}"; then&#xA;
+          python $SONIC_CLI_ROOT/sonic-cli-tam.py get_sonic_tam_sonic_tam_tam_collector_table show_tam_collector.j2&#xA;
+        else&#xA;
+          python $SONIC_CLI_ROOT/sonic-cli-tam.py get_sonic_tam_sonic_tam_tam_collector_table_tam_collector_table_list ${name} show_tam_collector.j2&#xA;
+        fi&#xA;
+      </ACTION>
+    </COMMAND>
+  </VIEW>
 
-         <COMMAND
-        name="no device-id"
-        help="Remove the Device Id of the box"
-        view="configure-tam-view"
-        >
-              <ACTION>
-               python $SONIC_CLI_ROOT/tam.py -clear -device -deviceid 0 
-              </ACTION>
-        </COMMAND>
+  <VIEW name="configure-view">
+    <!-- tam -->
+    <COMMAND name="tam" help="TAM Device Configuration" mode="subcommand" ptype="SUBCOMMAND" view="config-tam"/>
```

Comment:
Can we keep the name earlier name configure-tam-view instead of config-tam, as other tam features are using configure-tam-view for backward compatibility ?

Reply:
What is meant by backward compatibility here ? CLI commands are getting added via clish now.

Reply:
Broadcom is also adding commands for other features in the same tam view